Metallum/Classes/Controllers/NetworkManager.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  NetworkManager.swift
+//  Metallum
+//
+//  Created by AppleTree on 3/3/16.
+//  Copyright © 2016 OlhaF. All rights reserved.
+//
+
+import Foundation
+
+final class NetworkManager {
+    
+    static let sharedInstance = NetworkManager()
+    
+    let session: NSURLSession
+    var jsonDictionary: NSDictionary?
+    
+    private init() {
+        let config = NSURLSessionConfiguration.defaultSessionConfiguration()
+        config.timeoutIntervalForRequest = 30.0
+        config.timeoutIntervalForResource = 60.0
+        config.HTTPAdditionalHeaders = ["application-id" : "D65ECD64-92B7-4FC8-FF53-CC8CB9314200",
+                                        "secret-key" : "663FD35E-769A-3B6A-FF9D-7C3376D3C200",
+                                        "application-type" : "REST"]
+        
+        session = NSURLSession(configuration: config)
+    }
+    
+    func downloadData(complerionHandler completion: ()->Void) {
+        let resourceURL = NSURL(string: "https://api.backendless.com/v1/data/Band")
+        let currentRequest = NSURLRequest(URL: resourceURL!)
+        
+        let task = session.dataTaskWithRequest(currentRequest) { (let data, let response, let error) -> Void in
+            if ((response?.respondsToSelector("statusCode")) != nil) {
+                if let httpResponse = response as? NSHTTPURLResponse {
+                    if httpResponse.statusCode == 200 {
+                        self.parseJSONFromData(data!)
+                        if self.jsonDictionary != nil {
+                            completion()
+                        }
+                    } else {
+                        print("HTTPResponse status code - \(httpResponse.statusCode)")
+                    }
+                }
+            } else {
+                print("Error - doesn't respond to Status Code")
+            }
+            if error != nil {
+                print("Data task error - \(error)")
+            }
+            
+        }
+        task.resume()
+    }
+    
+    func parseJSONFromData(data: NSData) {
+        do {
+            let jsonDictionary = try NSJSONSerialization.JSONObjectWithData(data, options: .AllowFragments)
+            if let dict = jsonDictionary as? NSDictionary {
+                self.jsonDictionary = dict
+                //print(dict)
+            }
+        } catch {
+            print("Error during parsing - \(error)")
+        }
+    }
+    
+    func createBands(jsonDictionary: NSDictionary?) -> [Band]? {
```

Comment:
Creating  of bands is not duty of NetworkManager.
Use failable constructor that takes JSON in Band and Member model
In BandsViewController in completionHandler create array of band from array of JOSNs (use failable constructor)